### DISCUSSION TITLE
Delete the output files without closing matlab

### DISCUSSION THREAD

**Discussion by Peku995** [cdslaborg/paramonte#30] [Q&A] [answered]
Hi,

I can't delete the output files because it is stated that the files are stille opened in matlab after the simulation finished. is it possible to delete the output files without closing matlab?

Thanks

**Comment by fagheri** [comment, marked as answer]
Hi @Peku995 Under normal termination conditions, the file ownerships should be released automatically to the user. 
But only in Windows when the simulation fails for any reason then Windows keeps the file ownership with the failed simulation (i.e., with MATLAB). This can be fixed within the sampler implementation by enforcing file closure should the simulations fail at any stage. 
We will try to fix this in the impending release.
Thanks for reporting this issue! :-)

**Reply by Peku995** [reply]
Hi @fagheri , thanks for your answer. However, the simulations did not fail, at least not in a way that I would have notized it, but still I can't delete it.

**Comment by shahmoradi** [comment]
Hi @Peku995, as a temporary bypass for now, note that you can set the simulation variable [overwriteRequested](https://www.cdslab.org/paramonte/notes/usage/paradram/specifications/#overwriterequested) to a `true` value to  let the sampler automatically overwrite the files for every new run with the same file names. This should work. If it does not, note that you can always drop the simulation specification [outputFileName](https://www.cdslab.org/paramonte/notes/usage/paradram/specifications/#outputfilename) or end it with a `\` indicating a folder name to let the sampler create a new set of files in the current or specified folder, based on the current date and time for each new simulation. Once you close matlab for the day, you can delete the folder containing the files all together.
